pubsub/sqs_pub.go

```diff
@@ -56,13 +57,17 @@ func GetQueueUrl(client *sqs.SQS, topic string) string {
 	return *out.QueueUrl
 }
 
-// Publish sends an error message to the DLQ `topic`
-func (s *SqsPublisher) Publish(topic string) {
-	queueUrl := GetQueueUrl(s.client, topic)
-	s.logger = log.NewLog(fmt.Sprintf("SQS-Pub{%s}", topic))
-	s.logger.Info("SQS Publisher started for queue: %s", queueUrl)
-	for eventResponse := range s.errors {
+// Publish sends an error message to provided topics
+func (s *SqsPublisher) Publish(errorsTopic string, acksTopic string) {
+	s.logger.Info("SQS Publisher started for topics: %s %s", errorsTopic, acksTopic)
+	for eventResponse := range s.notifications {
 		delay := int64(0)
+
+		queueUrl := GetQueueUrl(s.client, errorsTopic)
```

Comment:
please really, really, don't do this
you're looking up the URL for your topic (and it's a full round-trip to AWS management APIs) for *every* message

Please look up and store the queue URL in TWO variables **outside** the `for` loop, and then pick the right one based on (a) is it an OK message and (b) are we publishing Ok messages